docs/spec_draft.md

```diff
@@ -305,7 +309,32 @@ the IEEE-754 specification. For boolean element type, the behavior is same as
 // %result: [[6, 8], [10, 12]]
 ```
 
-&nbsp;[More Examples](../stablehlo/tests/interpret_add.mlir)
+# stablehlo.after_all
+
+### Semantics
+Ensures that the operations producing the `inputs` are ordered before any
+operation that depend on `result`.
+
+### Inputs
+
+| Name     | Type                       |
+|----------|----------------------------|
+| `inputs` | variadic number of `token` |
+
+### Outputs
+
+| Name     | Type  |
+|----------|-------|
+| `result` | token |
+
+### Constraints
+  * size(`inputs`) $\ge$ 0.
```

Comment:
How would a size of inputs be less than zero? I think we can drop the entire Constraints section.